### ISSUE TITLE
Add Python 3.8 release candidate to our CI

### ISSUE THREAD

**Issue by Zac-HD** [HypothesisWorks/hypothesis#2102]
[We're in the final beta period and moving to release candidate soon](https://discuss.python.org/t/action-required-python-3-8-0b4-available-for-testing/2231), and CPython devs have asked the community to test against the beta while there's still time to make code changes.  So it would be nice to add it to our CI!

Out of scope for this issue, [there are a bunch of `typing` changes](https://docs.python.org/3.8/whatsnew/3.8.html#typing) which I'll need to handle in `st.from_type()`.  More seriously, positional-only parameters will demand some overhaul of our decorator logic for `@given` and argument analysis for `st.builds()`... but both of those should be considered blocked on getting a 3.8 CI job anyway, so let's do the simple thing first :slightly_smiling_face:
